SharedLibs/TileMapHelper/src/model/Model.ts

```diff
@@ -239,136 +277,17 @@ export class TileObject {
+   * the Pixi's rotate
    */
-  getTileId(): integer {
-    return FlippingHelper.getTileId(this.tileId);
-  }
-
-  setFlippedHorizontally(flippedHorizontally: boolean): void {
-    this.tileId = FlippingHelper.setFlippedHorizontally(
-      this.tileId,
-      flippedHorizontally
-    );
-  }
-
-  setFlippedVertically(flippedVertically: boolean): void {
-    this.tileId = FlippingHelper.setFlippedVertically(
-      this.tileId,
-      flippedVertically
-    );
-  }
-
-  setFlippedDiagonally(flippedDiagonally: boolean): void {
-    this.tileId = FlippingHelper.setFlippedDiagonally(
-      this.tileId,
-      flippedDiagonally
-    );
-  }
-
-  /**
-   * @returns true if the tile is flipped horizontally.
-   */
-  isFlippedHorizontally(): boolean {
-    return FlippingHelper.isFlippedHorizontally(this.tileId);
-  }
-
-  /**
-   * @returns true if the tile is flipped vertically.
-   */
-  isFlippedVertically(): boolean {
-    return FlippingHelper.isFlippedVertically(this.tileId);
-  }
-
-  /**
-   * @returns true if the tile is flipped diagonally.
-   */
-  isFlippedDiagonally(): boolean {
-    return FlippingHelper.isFlippedDiagonally(this.tileId);
-  }
-}
-
-/**
- * Tile identifiers making to access flipping flags.
- */
-class FlippingHelper {
-  static readonly flippedHorizontallyFlag = 0x80000000;
-  static readonly flippedVerticallyFlag = 0x40000000;
-  static readonly flippedDiagonallyFlag = 0x20000000;
-  static readonly tileIdMask = ~(
-    FlippingHelper.flippedHorizontallyFlag |
-    FlippingHelper.flippedVerticallyFlag |
-    FlippingHelper.flippedDiagonallyFlag
-  );
-
-  static getTileId(tileId: integer): integer {
-    return tileId & FlippingHelper.tileIdMask;
-  }
-
-  static setFlippedHorizontally(
-    tileId: integer,
-    flippedHorizontally: boolean
-  ): integer {
-    tileId &= ~FlippingHelper.flippedHorizontallyFlag;
-    if (flippedHorizontally) {
-      tileId |= FlippingHelper.flippedHorizontallyFlag;
-    }
-    return tileId;
-  }
-
-  static setFlippedVertically(
-    tileId: integer,
-    flippedVertically: boolean
-  ): integer {
-    tileId &= ~FlippingHelper.flippedVerticallyFlag;
-    if (flippedVertically) {
-      tileId |= FlippingHelper.flippedVerticallyFlag;
-    }
-    return tileId;
-  }
-
-  static setFlippedDiagonally(
-    tileId: integer,
-    flippedDiagonally: boolean
-  ): integer {
-    tileId &= ~FlippingHelper.flippedDiagonallyFlag;
-    if (flippedDiagonally) {
-      tileId |= FlippingHelper.flippedDiagonallyFlag;
-    }
-    return tileId;
-  }
-
-  static isFlippedHorizontally(tileId: integer): boolean {
-    return (tileId & FlippingHelper.flippedHorizontallyFlag) !== 0;
-  }
-
-  static isFlippedVertically(tileId: integer): boolean {
-    return (tileId & FlippingHelper.flippedVerticallyFlag) !== 0;
-  }
-
-  static isFlippedDiagonally(tileId: integer): boolean {
-    return (tileId & FlippingHelper.flippedDiagonallyFlag) !== 0;
-  }
-}
+  readonly rotate: integer;
```

Comment:
It's an extra int: `\"Flip bits\", a 2-bits integer to represent the mirror transformations of the tile.<br/>   - Bit 0 = X flip<br/>   - Bit 1 = Y flip<br/>   Examples: f=0 (no flip), f=1 (X flip only), f=2 (Y flip only), f=3 (both flips)`.
I think it will be the most controversial change.
Since the bit flips were occurring in some many times, I think it's better to directly save the results.

Reply:
> It's an extra int: `\"Flip bits\", a 2-bits integer to represent the mirror transformations of the tile.<br/> - Bit 0 = X flip<br/> - Bit 1 = Y flip<br/> Examples: f=0 (no flip), f=1 (X flip only), f=2 (Y flip only), f=3 (both flips)`.

Ok, so tiled only do flipping where Tile use a 3rd bit to handle rotation.
As LDTK has a lower level of feature as Tiled for flipping/rotation, the model doesn't need a new attribute for it. The diagonal flipping bit can be filled with a 0 by the `Loader` (using the right model method that encapsulates the masking).

> Since the bit flips were occurring in some many times, I think it's better to directly save the results.
Using a tile identifier that is only a number is actually more efficient to map identifiers to their texture when rendering. It also has the advantage makes tile map memory representations lighter and sequential.

Reply:
I forgot... the final push for the change was the support for the stacked tiles.
Instead of `Array<Int32Array>`, it would need to be a `Array<Array<Int32Array>>`. All the nice optimizations will go to the gutters.

My approach use a bit more memory but it use less cpu and the code is more clearer, I think.

Reply:
I thought of a tuple like `[diagonal: boolean, horizontal: boolean, vertical: boolean, tileId1: integer, rotate1: integer, tileId2: integer, rotate2: integer, ...]` with just array but the code isn't pretty and easier to read...

Reply:
> I forgot... the final push for the change was the support for the stacked tiles. Instead of `Array<Int32Array>`, it would need to be a `Array<Array<Int32Array>>`. All the nice optimizations will go to the gutters.
> 
> My approach use a bit more memory but it use a bit less cpu and the code is more clearer, I think.

What do you think about defining tile stacking in the tile set?
A stack could be seen as a tile definition and we can assign them an id like any tile. It will allow only 512Mi combinations but it should be enough as every possible combination won't be used. This way we can keep the optimization and reduce the scope of this PR changes.

Reply:
I will make the changes

Reply:
I'm not a fan to have to compute the true tileId (from the tileId+flipping bits) for every tiles, every time we are drawing the tile map.
I find that extra computing to be a bit painful just for a bit less memory.

No problem to use the tile definition for the stacked tiles.

Reply:
The last commit is using bit flipping.